### PR TITLE
Implement new design for Review screen of User Input page with the new Inline Edit interface.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #5891 

## Relevant technical choices

- Render `UserInputSelectOptions` in `UserInputPreviewGroup` rather than `UserInputQuestionWrapper`, which was not needed.
- Maybe the IB was a bit out of date, there was no `hasInvalidValues` function. It was necessary to refactor the error handling a bit to support the requirements.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by asvinb** [comment]
Nice job @techanvil ! LGTM!

File: assets/js/components/user-input/UserInputSelectOptions.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ export default function UserInputSelectOptions( {
 	max,
 	next,
 	isActive,
+	noInstructions,
```
Comment (tofumatt) [inline assets/js/components/user-input/UserInputSelectOptions.js:46]:
```suggestion
	noInstructions,
```

I suspect the reason this was used is the slightly nicer API of `<UserInputSelectOptions noInstructions />` over `<UserInputSelectOptions instructions={ false } />`. But the double-negative, especially in the code below:

```js
			{ ! noInstructions && (
				<p className="googlesitekit-user-input__select-instruction">
					<span>
						{ sprintf(
							/* translators: %s: number of answers allowed. */
							_n(
								'Select only %d answer',
								'Select up to %d answers',
								max,
								'google-site-kit'
							),
							max
						) }
					</span>
				</p>
			) }
```

is harder to parse than:

```js
			{ showInstructions && (
				<p className="googlesitekit-user-input__select-instruction">
					<span>
						{ sprintf(
							/* translators: %s: number of answers allowed. */
							_n(
								'Select only %d answer',
								'Select up to %d answers',
								max,
								'google-site-kit'
							),
							max
						) }
					</span>
				</p>
			) }
```

Or maybe even `hideInstructions` would be a better variable name, as it's a bit more of an instructive prop name to be used than setting a negatively-named boolean to `true`. It's just a bit roundabout to me. 😅

**Review by tofumatt — CHANGES_REQUESTED** [review]
Code looks great and the functionality seems good, just one _incredibly minor_ point and then we can merge this, nice 👍🏻

File: assets/js/components/user-input/UserInputSelectOptions.js (hunk 1 above)
Comment (techanvil) [reply assets/js/components/user-input/UserInputSelectOptions.js:46]: Hey @tofumatt, thanks for the comment. I was, in part, taking my cue from other components, for example:

https://github.com/google/site-kit-wp/blob/af97300672d201e874227729b02ac009ae20791c/assets/js/components/user-input/UserInputPreview.js#L113-L133

However, I agree the double negative is not so nice to read, and am happy to change it to `showInstructions`.

**Comment by techanvil** [comment]
> Code looks great and the functionality seems good, just one _incredibly minor_ point and then we can merge this, nice 👍🏻
> 
> One thing to note is there's a LOT of VRT failures here, I wonder if it's a bug on CI?
> 
> Lots of unrelated changes like:
> 
> CleanShot 2022-12-13 at 22 18 12
> 
> CleanShot 2022-12-13 at 22 18 12
> 
> Maybe it's a temporary thing and a new run'll fix it? 🤷🏻‍♂️

Thanks @tofumatt! I think the VRT thing is simply due to the PR being a bit old and reflecting the broken state of the VRT workflow at the time. I've merged `develop` in, hopefully that will resolve things here.

File: assets/js/components/user-input/UserInputSelectOptions.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/components/user-input/UserInputSelectOptions.js]: Ah fair point. The struggle between consistency and partial improvements 😅

Thanks, I think this is all good now 👍🏻

**Pull request merged**